data/common.yaml

```diff
@@ -9,67 +9,3 @@ lookup_options:
     merge: unique
 
 mlocate::locate: mlocate
-
-mlocate::prunenames:
-  - .git
-  - .hg
-  - .svn
-
-mlocate::prunepaths:
-  - /afs
-  - /media
-  - /net
-  - /sfs
-  - /tmp
-  - /udev
-  - /var/cache/ccache
-  - /var/spool/cups
-  - /var/spool/squid
-  - /var/tmp
-
-mlocate::prunefs:
-  - 9p
-  - afs
-  - anon_inodefs
-  - auto
-  - autofs
-  - bdev
-  - binfmt_misc
-  - cgroup
-  - cifs
-  - coda
-  - configfs
-  - cpuset
-  - debugfs
-  - devpts
-  - ecryptfs
-  - exofs
-  - fuse
-  - fusectl
-  - gfs
-  - gfs2
-  - gpfs
-  - hugetlbfs
-  - inotifyfs
-  - iso9660
-  - jffs2
-  - lustre
-  - mqueue
-  - ncpfs
-  - nfs
-  - nfs4
-  - nfsd
-  - pipefs
-  - proc
-  - ramfs
-  - rootfs
-  - rpc_pipefs
-  - securityfs
-  - selinuxfs
-  - sfs
-  - sockfs
-  - sysfs
-  - tmpfs
-  - ubifs
-  - udf
-  - usbfs
```

Comment:
Making this system-dependent means that the same catalog on different nodes produce a different result, AFAIAC, this looks like a smell because I expect consistent configuration across nodes using the same Puppet configuration regardless of the OS they are running.  Some modules already do something similar (e.g. puppet-fail2ban).  I do not know if I am the only one worried by this, maybe we should discuss this at some office-hours?

Reply:
Interesting - clearly I've tried to emulate the default OS configurations as a default.

If you then add an extra prunepath of `/var/cache/squid` that will work on all OSes de-duping it if needed. Loosing the suggestions of the OS seems a loss when say migrating from yum to dnf for instance.

For me puppet should configure the OS beyond what the OS provides and not define the OS.
Lots of examples elsewhere I would say, Apache , rsyslog.